### ISSUE TITLE
 no data

### ISSUE THREAD

**Issue by Yarulik** [wlmeng11/SimpleRick#1]
Hello William!
I repeated your device, flashed it, but there is no data, the servo does not move. Even after pressing a button. As I understand it, my calibration does not pass. Do you have suggestions?
HV pulser is working, servo line is working. I am using a piezo with a resonant frequency of 2 and 3 MHz.

**Comment by Yarulik** [comment]
good news
reconnecting usb sometimes writes the end of the calibration in the terminal.  from the generator the signal is amplified to VGA, TGC works.  By pressing the button, the servomotor does not move, the software shows the redundancy of signals from the generator and without.

**Comment by wlmeng11** [comment]
Hello! I'm assuming you are using my firmware from here? https://github.com/wlmeng11/SimpleRick_firmware
If you look at the Serial Monitor in Arduino, do you see any messages once you press the button? It should show some messages like "Starting image sweep"

**Comment by wlmeng11** [comment]
Does the servo work with other microcontrollers? Just to make sure it isn't faulty.

**Comment by Yarulik** [comment]
Yes, the servo is working, it rises to the zero position. Now he writes Start image scan, sometimes he first writes calibration completion and no longer writes "Start image scan". Yes I am using your firmware

**Comment by Yarulik** [comment]
For some reason, after flashing the Arduino IDE in Windows, it started displaying messages! What is the Teensy processor frequency?

**Comment by wlmeng11** [comment]
Ah fantastic! I don't remember exactly the Teensy frequency but I believe it was the default setting of 96 MHz.

**Comment by wlmeng11** [comment]
The correct clock frequency is important for producing the correct delays for the HV pulse signal, since the pulse width is determined by the period of the Teensy clock (by using the NOP assembly instruction). 96MHz frequency = 10.4ns period. In my code, I produced a 50ns pulse width with 3 NOP commands (3x10.4ns) plus the rise and fall times (each is on the order of 10ns).

**Comment by Yarulik** [comment]
I recently started studying ultrasound technology based on open sources and based on the work of kelu124. As it turns out, you William made an excellent debug board. This reduced design time. In the future, I plan to make underwater frontal sonar for ROV with a resonance frequency of up to 2 Mhz
I did not understand how to set, change the resonant frequency for a piezo element.

**Comment by Yarulik** [comment]
> Ah fantastic! I don't remember exactly the Teensy frequency but I believe it was the default setting of 96 MHz.

like after that and he started to write in the terminal. Fine! this already pleases)

**Comment by wlmeng11** [comment]
Great! Let me know if you run into any other issues with the board!
You can't change the resonant frequency of the piezo element because it is determined by the mechanical dimensions of the piezo. If you're asking about how to change the receiver frequency on the RTL SDR to match the piezo frequency, you can use the `-f` parameter in the `rtl_to_npz` script. https://github.com/wlmeng11/rtl-ultrasound/blob/c9e91015478f502ef04b457377ec40419ce96129/bin/rtl_to_npz#L33

**Comment by Yarulik** [comment]
a warning is issued, before loading the firmware warning: warning: 
format '% d' expects argument of type 'int *', but argument 3 has type 'int' [-Wformat =]
